### PR TITLE
#22899 : Contentlets inside the block editor not being push published.

### LINKED ISSUE #22899: Block editor: Contentlets inside the block editor not being push published
**Describe the bug**
When you push publish a contentlet that have a block editor field with a contentlet in it, the contentlet is not pushed.

**To Reproduce**
1. Create a content type with a block editor
2. Create a contentlet and in the block editor add a contentlet or image block
3. Push publish that contentlet
4. See in the server, the image broken.

**Expected behavior**
It should push the related content inside the block editor.

**Additional context**
- We need to make sure this works with images and contentlets in the block editor.
- Make sure the related content doesn't exist in the destination

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
### Proposed Changes
* Contentlets and dotAssets must be push published when they're referenced inside a Story Block field via Push Publishing Dependencies. The Story Block API allows developers to easily interact with data in this type of field.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/publisher/util/dependencies/PushPublishigDependencyProcesor.java [hunk 1]
```diff
@@ -427,14 +441,30 @@ private void processContentDependency(final Contentlet contentlet)
...
+                                    this.contentletAPI.get().findContentletByIdentifier(contentletId,
+                                            contentletVersion.isLive(), contentletVersion.getLanguageId(),
+                                            APILocator.systemUser(), false);
+                            tryToAddAndProcessDependencies(PusheableAsset.CONTENTLET, contentInStoryBlock,
+                                    ManifestReason.INCLUDE_DEPENDENCY_FROM.getMessage(contentletVersion));
+                        } catch (final DotDataException | DotSecurityException e) {
+                            Logger.warn(this,
+                                    String.format("Could not analyze dependent Contentlet '%s' referenced " + "in " + "Story Block field from Contentlet Inode " + "'%s': %s", contentletId, contentletVersion.getInode(), e.getMessage()));
+                        }
+                    });
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/publisher/util/dependencies/PushPublishigDependencyProcesor.java:466]: maybe we can move this to a local method something like: processStoryBlock(...)

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/publisher/util/dependencies/PushPublishigDependencyProcesor.java (hunk 1 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/publisher/util/dependencies/PushPublishigDependencyProcesor.java:466]: yeah

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/publisher/util/dependencies/PushPublishigDependencyProcesor.java (hunk 1 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/publisher/util/dependencies/PushPublishigDependencyProcesor.java]: Done!

**Comment by fmontes** [comment]
Needs to point this to 22.11

**Pull request closed without merging**